common/pc/laptop/battery.nix

```diff
@@ -19,11 +24,33 @@ in
       type = lib.types.bool;
     };
   };
+  imports = (
+    map
+      (
+        option:
+        lib.mkRemovedOptionModule [
+          "hardware"
+          "asus"
+          "battery"
+          option
+        ] "The hardware.asus.battery.* options were removed in favor of `hardware.battery.*`."
+      )
+      [
+        "chargeUpto"
+        "enableChargeUptoScript"
+      ]
+  );
   config = {
     environment.systemPackages = lib.mkIf cfg.enableChargeUptoScript [ p ];
     systemd.services.battery-charge-threshold = {
-      wantedBy = [ "local-fs.target" "suspend.target" ];
-      after = [ "local-fs.target" "suspend.target" ];
+      wantedBy = [
+        "local-fs.target"
+        "suspend.target"
+      ];
+      after = [
+        "local-fs.target"
+        "suspend.target"
+      ];
       description = "Set the battery charge threshold to ${toString cfg.chargeUpto}%";
       startLimitBurst = 5;
       startLimitIntervalSec = 1;
```

Comment:
The script below currently only works with one battery, right?
I know that some thinkpads have a soldered battered and a changeable one.

Reply:
Are you sure this interface is actually portable? I see quite a few different approaches here: https://linrunner.de/tlp/settings/bc-vendors.html
There don't seem to be a standard.

Reply:
Ok. Looks like `charge_control_end_threshold` is the most common interface though.

Reply:
Right ya not completely portable. I'm happy to update the script to test if that interface exists and then print a warning and exit if not. Then I (or others as needed over time) could add support for other battery interfaces. Can also put a note in the option about the supported interfaces. But then there is still the issue that inevitably some people will import/set it and not get their expected behavior I guess. 

Open to other suggestions, or if you don't think it's worth pursuing, feel free to close.

Reply:
Probably failing the service as it does now, is probably the best course of action.
However it would be cool if you could link to https://linrunner.de/tlp/settings/bc-vendors.html in the code so that people can figure out themselves if there battery is not supported.

Reply:
Also I would change it to iterate over all batteries instead of using just one.

Reply:
> Also I would change it to iterate over all batteries instead of using just one.

The original command already does echo into a folder with `/BAT?/` so will update all of the batteries.

I did update the code to only declare the interface path once, mention that it applies to all batteries, and link the vendor page and also suggest tlp package if the script isn't applicable. 

I didn't get a chance to test the latest commit on a real asus laptop yet, but it does pass the test runs.